model/legacy-private/src/main/java/org/keycloak/models/dblock/DBLockLockingMechanismProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.dblock;
+
+import org.keycloak.models.locking.LockingMechanismProvider;
+
+public class DBLockLockingMechanismProvider implements LockingMechanismProvider {
+
+    private final DBLockProvider dbLockProvider;
+    public DBLockLockingMechanismProvider(DBLockProvider dbLockProvider) {
+        this.dbLockProvider = dbLockProvider;
+    }
+
+    private static DBLockProvider.Namespace stringToNamespace(String lockName) {
+        if ("database".equals(lockName)) {
+            return DBLockProvider.Namespace.DATABASE;
+        } else if ("keycloak-boot".equals(lockName)) {
+            return DBLockProvider.Namespace.KEYCLOAK_BOOT;
+        } else if ("offline-sessions".equals(lockName)) {
+            return DBLockProvider.Namespace.OFFLINE_SESSIONS;
+        } else {
+            throw new RuntimeException("Lock with name " + lockName + " not supported by DBLockLockingMechanism");
+        }
+    }
+
+    @Override
+    public void acquireLock(String lockName) {
```

Comment:
One question: I didn't replace the usage of `DBLock` in legacy modules. For example, I left it [here](https://github.com/keycloak/keycloak/blob/9c007e3779202b1516d4495c49494e2701369f39/model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/initializer/DBLockBasedCacheInitializer.java#L69). Do you think it makes sense to use the new `GlobalLock` in legacy modules? To me, it seems unnecessary as all legacy code will always use only `DBLock` at the end.

Reply:
I agree, legacy can stay as is